leo/main.rs

```diff
@@ -175,8 +175,38 @@ enum CommandOpts {
     // },
 }
 
+fn set_panic_hook() {
+    #[cfg(not(debug_assertions))]
+    std::panic::set_hook({
+        Box::new(move |e| {
+            println!(
+                "thread `{}` {}",
+                std::thread::current().name().unwrap_or("<unnamed>"),
+                e
+            );
+            println!("stack backtrace: \n{:?}", backtrace::Backtrace::new());
+            println!("error: internal compiler error: unexpected panic\n");
+            println!("note: the compiler unexpectedly panicked. this is a bug.\n");
+            println!("note: we would appreciate a bug report: https://github.com/AleoHQ/leo/issues/new?labels=bug,panic&template=bug.md&title=[Bug]\n");
+            println!(
+                "note: {} {} running on {} {}\n",
+                env!("CARGO_PKG_NAME"),
+                env!("CARGO_PKG_VERSION"),
+                sys_info::os_type().unwrap_or_else(|e| e.to_string()),
+                sys_info::os_release().unwrap_or_else(|e| e.to_string()),
```

Comment:
Is it possible to find out the os without using an external `sys-info` crate?
Potentially from env variables https://doc.rust-lang.org/std/env/consts/constant.OS.html?

Reply:
You can get the OS type without `sys-info` but not the version of the OS iirc.

Reply:
We're not using the advanced features of `sys-info`. Is there a flag we can use on import to disable those features?

Reply:
no new dependencies from `sys-info` it is fine to import